transport.go

```diff
@@ -1,15 +1,28 @@
-// Deprecated: This package has moved into go-libp2p as a sub-package: github.com/libp2p/go-libp2p/p2p/muxer/mplex.
-package peerstream_multiplex
+package mplex
```

Comment:
Should we keep the package-level deprecation?

Reply:
Yeah. We should.

Reply:
https://github.com/libp2p/go-libp2p-mplex/pull/38/commits/ccfff8377dedde91c6b74ed5e1db57e741471ee6